tests/test_causal_model.py

```diff
@@ -275,5 +275,120 @@ def test_graph_input4(self, beta, num_instruments, num_samples, num_treatments):
         all_nodes = model._graph.get_all_nodes(include_unobserved=False)
         assert "Unobserved Confounders" not in all_nodes
 
+    @pytest.mark.parametrize(["beta", "num_instruments", "num_samples"],
+                             [(10, 2, 5000),])
+    def test_graph_refutation(self, beta, num_instruments, num_samples):
+        data = dowhy.datasets.linear_dataset(beta=beta,
+        num_common_causes=5,
+        num_instruments = num_instruments,
+        num_effect_modifiers=1,
+        num_samples=num_samples, 
+        treatment_is_binary=True,
+        stddev_treatment_noise=10,
+        num_discrete_common_causes=1)
+        df = pd.read_csv('tests/test_data.csv')
+        model = CausalModel(
+            data=df,
+            treatment=data["treatment_name"],
+            outcome=data["outcome_name"],
+            graph=data["gml_graph"],
+            proceed_when_unidentifiable=True,
+            test_significance=None
+        )
+        gml_str = """dag {
+        W0 [pos="-2.200,-1.520"]
+        W1 [pos="-1.457,-1.533"]
+        W2 [pos="-0.763,-1.547"]
+        W3 [pos="1.041,-1.587"]
+        W4 [pos="1.510,-1.560"]
+        X0 [pos="1.222,-0.625"]
+        Z0 [pos="0.390,-1.601"]
+        Z1 [pos="-0.176,-1.540"]
+        v0 [pos="-0.219,-0.881"]
+        y [pos="-0.144,-0.296"]
+        W0 -> v0
+        W0 -> y
+        W1 -> v0
+        W1 -> y
+        W2 -> v0
+        W2 -> y
+        W3 -> v0
+        W3 -> y
+        W4 -> v0
+        W4 -> y
+        X0 -> y
+        Z0 -> v0
+        Z1 -> v0
+        v0 -> y
+        }"""
+        model = CausalModel(
+            data=df,
+            treatment=data["treatment_name"],
+            outcome=data["outcome_name"],
+            graph=gml_str,
+            proceed_when_unidentifiable=True,
+            test_significance=None,
+            missing_nodes_as_confounders=True
+        )
+        graph_refuter_object = model.refute_graph(1,method_name = "partial_correlation")
+        assert graph_refuter_object._refutation_passed == True
+        assert len(graph_refuter_object._true_implications) > 0
+
+    @pytest.mark.parametrize(["beta", "num_instruments", "num_samples"],
+                             [(10, 2, 5000),])
+    def test_graph_refutation2(self, beta, num_instruments, num_samples):
+        data = dowhy.datasets.linear_dataset(beta=beta,
+        num_common_causes=5,
+        num_instruments = num_instruments,
+        num_effect_modifiers=1,
+        num_samples=num_samples, 
+        treatment_is_binary=True,
+        stddev_treatment_noise=10,
+        num_discrete_common_causes=1)
+        df = pd.read_csv('tests/test_data.csv')
```

Comment:
is there a way to generate the data using code? You can set a random seed so that the data is always the same from linear_dataset. Would like to avoid adding another file (and these kind of tests always break due to folder location referencing issues)